### ISSUE TITLE
Getting a PHP warning for Illegal string offset 'url'

### ISSUE THREAD

**Issue by bmarshall511** [dudaster/ele-custom-skin#118]
Created a loop template using the Flip box widget and a dynamic url for the link. Getting: 

```
Warning
: Illegal string offset 'url' in
.....wp-content/plugins/elementor/includes/base/controls-stack.php
on line 1177
```

I've used other Flip box widgets elsewhere outside of Skin and work fine. Seems to only happen when using a look template.

Ex. https://rebuild-sparkcognition.pantheonsite.io/

**Comment by dudaster** [comment]
I know the issue. I'm still looking for a solution, even asked Elementor guys to have in mind this issue. 

If you are using ECS Pro just add {{permalink}} in the link field and turn on Dynamic Everywhere in Posts Widget.

**Comment by fafchook** [comment]
Having the same exact issue. 
Is there a solution in the near future?

**Comment by digitayfun** [comment]
I am having the same issue 

Warning Illegal string offset 'url' in  wp-content/plugins/elementor/includes/base/controls-stack.php  on line  1177

**Comment by maximseshuk** [comment]
I am having the same issue if use Control Type URL dynamic field data.

Warning:  Illegal string offset 'url' in wp-content\plugins\elementor\includes\base\controls-stack.php on line 1177

**Comment by ambertripp1983** [comment]
I'm having the exact same issue.  Any luck finding a solution?

**Comment by maximseshuk** [comment]
> I'm having the exact same issue. Any luck finding a solution?

This is an Elementor problem. I found a solution in Elementor code, but I'm not sure if it's perfect.

**Comment by ambertripp1983** [comment]
I'll take what I can get for now just to get some functions working.  Where in Elementor did you find the solution?

**Comment by dudaster** [comment]
You need to hack elementor's code and I wouldn't recomend, you'll loose the changes with elementor updates.

If you are a ECS Pro user you can use dynamic everywhere to workaround the problem by typing {{permalink}} in the custom URL field.

**Comment by TarunNarula** [comment]
I resolved it for a temporary solution. Changing line no. 176 from

`if ( $dynamic_property ) ) {`
to
`if ( $dynamic_property && isset($settings[ $control_name ][ $dynamic_property ]) ) {`

**Comment by dudaster** [comment]
no. 176 from what file?

**Comment by pippyn** [comment]
I found it too.. And can confirm this temporary solution works.

It's line 1176 of /elementor/includes/base/controls-stack.php

**Comment by pippyn** [comment]
@dudaster Do you know if it's possible to fix this issue?

**Comment by dwillitzer** [comment]
I can confirm that modification does resolve the issue; had this issue come up on several sites.

**Comment by timaxsos** [comment]
Yes the fix works.

**Comment by PPjev** [comment]
Yes, the FIX **works**, it is on line 1204 now, latest version of elementor as of today.

**Comment by gbonamie** [comment]
I usually use this fix too -thanks @TarunNarula - but could not get it working after elementor's latest update -3.0.6.

UPDATE: works again with 3.0.7

**Comment by meilinda01** [comment]
I have the same issue

**Comment by EmerKelly** [comment]
@dudaster how would I get around this using the Progress Bar widget? I am using ECS Pro, so I can turn on Dynamic Everywhere, just not sure where to add {{permalink}}?
